vignettes/datatable-faq.Rmd

```diff
@@ -460,6 +460,70 @@ Happily, an internet search for "How does R method dispatch work" (at the time o
 
 However, features like basic S3 dispatch (pasting the function name together with the class name) is why some R folk love R. It's so simple. No complicated registration or signature is required. There isn't much needed to learn. To create the `merge` method for data.table all that was required, literally, was to merely create a function called `merge.data.table`.
 
+## Why do `T` and `F` behave differently from `TRUE` and `FALSE` in some `data.table` queries?
+
+Using `T` and `F` as abbreviations for `TRUE` and `FALSE` in `data.table` can lead to unexpected behavior. This is because `T` and `F` are global variables that can be redefined, which causes them to be treated as variable names rather than logical constants. This issue does not occur with `TRUE` and `FALSE`.
+
+To reproduce this issue:
+
+```r
+library(data.table)
+
+# Create a sample data.table
+DT <- data.table(x = rep(c("a", "b", "c"), each = 3), y = c(1, 3, 6), v = 1:9)
+
+# Using TRUE/FALSE works as expected
+# This selects the first two columns (x and y) and excludes the third one (v)
+DT[, .SD, .SDcols = c(TRUE, TRUE, FALSE)]
+#>    x y
+#> 1: a 1
+#> 2: a 3
+#> 3: a 6
+#> 4: b 1
+#> 5: b 3
+#> 6: b 6
+#> 7: c 1
+#> 8: c 3
+#> 9: c 6
+
+# Using T/F may lead to unexpected behavior
+# `data.table` treats T and F as variable names, not logical constants
```

Comment:
In comments `data.table` won't be rendered as code block so you can stick with data.table like you did in line 472 above